File: problems/002-two-sum/swift/clean.swift
import Foundation

func clean(_ c: [String: Any]) -> Any {
    let nums = (c["nums"] as! [Any]).map { ($0 as! NSNumber).intValue }
    let target = (c["target"] as! NSNumber).intValue
    var seen = [Int: Int]()
    for (i, n) in nums.enumerated() {
        if let j = seen[target - n] { return [j, i] }
        seen[n] = i
    }
    return [Int]()
}

File: problems/002-two-sum/kotlin/Clean.kt
class Clean : Solution {
    override fun solve(c: Map<String, Any?>): Any? {
        @Suppress("UNCHECKED_CAST")
        val nums = c["nums"] as List<Long>
        val target = c["target"] as Long
        val seen = HashMap<Long, Int>()
        nums.forEachIndexed { i, n ->
            seen[target - n]?.let { return listOf(it, i) }
            seen[n] = i
        }
        return emptyList<Int>()
    }
}
